src/server/dfly_main.cc

```diff
@@ -397,6 +403,10 @@ bool RunEngine(ProactorPool* pool, AcceptServer* acceptor) {
       LOG(ERROR) << "Could not open port " << port << ", error: " << ec.message();
       exit(1);
     }
+
+    if (port == 0) {
+      absl::SetFlag(&FLAGS_port, main_listener->socket()->LocalEndpoint().port());
```

Comment:
why do you need to set the flag?

Reply:
Some things (like replication, `INFO` command) report the port that dragonfly listens on